src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -1874,21 +1867,24 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
 
         private void VisitFieldAccessInternal(BoundExpression receiverOpt, FieldSymbol fieldSymbol)
         {
-            if (MayRequireTracking(receiverOpt, fieldSymbol) || (object)fieldSymbol != null && fieldSymbol.IsFixed)
+            bool asLvalue = (object)fieldSymbol != null &&
+                (fieldSymbol.IsFixed ||
+                !fieldSymbol.IsStatic && fieldSymbol.ContainingType.TypeKind == TypeKind.Struct);
```

Comment:
>!fieldSymbol.IsStatic && fieldSymbol.ContainingType.TypeKind == TypeKind.Struct [](start = 16, length = 79)

It doesn't look like this captures all relevant checks from MayRequireTracking. For example:
- ```receiverOpt.Kind != BoundKind.TypeExpression```
- ```!receiverOpt.Type.IsPrimitiveRecursiveStruct()```

Reply:
Fixed.